flocker/node/testtools.py

```diff
@@ -226,5 +226,5 @@ def to_node(node_state):
     :return Node: Equivalent node.
     """
     return Node(uuid=node_state.uuid, hostname=node_state.hostname,
-                applications=node_state.applications,
-                manifestations=node_state.manifestations)
+                applications=node_state.applications or [],
+                manifestations=node_state.manifestations or {})
```

Comment:
Unfortunately it's still perfectly valid to have missing applications or missing manifestations. Lots of tests require both because we have the stupid workaround for lack of leasing, but that isn't in place in many places.